pkg/url/url.go

```diff
@@ -8,32 +8,52 @@ import (
 	applabels "github.com/redhat-developer/odo/pkg/application/labels"
 	componentlabels "github.com/redhat-developer/odo/pkg/component/labels"
 	"github.com/redhat-developer/odo/pkg/occlient"
+	"github.com/redhat-developer/odo/pkg/util"
 
 	log "github.com/sirupsen/logrus"
 )
 
+// urlSourceNameAnnotation indicates the annotation used for the actual URL name (non metadata: name)
+const urlSourceNameAnnotation = "app.kubernetes.io/component-name"
+
 type URL struct {
 	Name     string
+	Service  string
 	URL      string
 	Protocol string
 }
 
 // Delete deletes a URL
-func Delete(client *occlient.Client, name string) error {
-	return client.DeleteRoute(name)
+func Delete(client *occlient.Client, urlName string, applicationName string) error {
+
+	// Namespace the URL name
+	namespacedOpenShiftObject, err := util.NamespaceOpenShiftObject(urlName, applicationName)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create namespaced name")
+	}
+
+	return client.DeleteRoute(namespacedOpenShiftObject)
 }
 
 // Create creates a URL
 func Create(client *occlient.Client, componentName, applicationName string) (*URL, error) {
 	labels := componentlabels.GetLabels(componentName, applicationName, false)
 
-	route, err := client.CreateRoute(componentName, labels)
+	// Namespace the component
+	namespacedOpenShiftObject, err := util.NamespaceOpenShiftObject(componentName, applicationName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create namespaced name")
+	}
+
+	// Pass in the namespace name, link to the service (componentName) and labels to create a route
+	route, err := client.CreateRoute(namespacedOpenShiftObject, labels)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to create route")
 	}
 
 	return &URL{
 		Name:     route.Name,
+		Service:  route.Labels[urlSourceNameAnnotation],
```

Comment:
What is `Service`?
Is it the name of the Service that this Route points to? Or is it the Component name that this URL points to?

Reply:
Service names aren't namespaced but the metadata name is. So this is needed to differentiate between Name and Service.

The service that the route points to will be the ComponentName

Reply:
I'm going to make some changes and see if it works.